### ISSUE TITLE
cmd/go: go:embed errors attributed to the wrong file

### ISSUE THREAD

**Issue by zombiezen** [golang/go#43469]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16beta1 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Haven't tried on tip.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/light/.cache/go-build"
GOENV="/home/light/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/light/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/light"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/light/sdk/go1.16beta1"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/light/sdk/go1.16beta1/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16beta1"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/light/src/foo/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build4010338713=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

Run `go build` on:

https://play.golang.org/p/HR-SdE2vE8c

### What did you expect to see?

```
bar/bar.go:5:1: pattern xxx.txt: no matching files found
```

### What did you see instead?

The line number is for the import of the package with the `go:embed` directive:

```
foo.go:5:2: pattern xxx.txt: no matching files found
```

**Comment by zombiezen** [comment]
Further, running `go build ./bar` outputs no line numbers:

```
pattern xxx.txt: no matching files found
```

I would expect to see the same output as what I expected above.

**Comment by bcmills** [comment]
Looks like there are no `cmd/go` tests for the `no matching files found` and `invalid pattern syntax` error messages, and the existing tests for the adjacent `pattern …: cannot embed …` errors lack anchors — so they aren't testing for the correct file or line numbers. 😞

**Comment by bcmills** [comment]
This is a release blocker for 1.16: `go:embed` is new in 1.16, and misattributing the file may mean that we're also misattributing the module, which could get very confusing for transitive errors in modules deep in the module cache.

**Comment by jayconrod** [comment]
The fix for this looks pretty small. We just need to resolve embeds a little lower in `Package.load`, after we push the path onto the import stack. That distinguishes errors in the importing and imported packages.

I'll wait a bit to send a fix until we decide how to resolve #43632, since that will affect tests.
